src/ConnectedMode/Embedded/SonarLintTargets.xml

```diff
@@ -20,11 +20,14 @@
 
     <PropertyGroup Condition="$(BuildingInsideVisualStudio) == 'true'">
         <SolutionDirAsInVs>$([System.IO.Path]::GetDirectoryName($(SolutionDir)))</SolutionDirAsInVs>
+        <SonarLanguage>$(Language)</SonarLanguage>
+        <SonarLanguage Condition=" $(Language) == 'C#' OR $(Language) == 'c#'">csharp</SonarLanguage>
+        <SonarLanguage Condition=" $(Language) == 'VISUALBASIC'">vb</SonarLanguage>
         <!-- C#/VB.NET settings need to be in different folders -->
-        <_SLVSRootFolder>$(APPDATA)\SonarLint for Visual Studio\Bindings\$(SolutionName)_$(SolutionDirAsInVs.GetHashCode())\$(Language)</_SLVSRootFolder>
+        <_SLVSRootFolder>$(APPDATA)\SonarLint for Visual Studio\Bindings\$(SolutionName)_$(SolutionDirAsInVs.GetHashCode())</_SLVSRootFolder>
 
-        <_SLVSGeneratedRoslynConfigFile>$(_SLVSRootFolder)\slvs_generated.globalconfig</_SLVSGeneratedRoslynConfigFile>
-        <_SLVSGeneratedAdditionalFile>$(_SLVSRootFolder)\SonarLint.xml</_SLVSGeneratedAdditionalFile>
+        <_SLVSGeneratedRoslynConfigFile>$(_SLVSRootFolder)\$(SonarLanguage).globalconfig</_SLVSGeneratedRoslynConfigFile>
+        <_SLVSGeneratedAdditionalFile>$(_SLVSRootFolder)\$(SonarLanguage)\SonarLint.xml</_SLVSGeneratedAdditionalFile>
```

Comment:
Technically, I think that SonarLanguage can be CSHARP whereas the folder on disk will be csharp. We had a comment in the code about this issue https://github.com/SonarSource/sonarlint-visualstudio/issues/1068
But I don't think it's relevant here, since this code is not supposed to run in MSBuild builds.